sdk/python/kfp/dsl/component_factory.py

```diff
@@ -35,6 +35,7 @@
 from kfp.dsl.types import type_utils
 
 _DEFAULT_BASE_IMAGE = 'python:3.7'
+SINGLE_OUTPUT_NAME = 'Output'
```

Comment:
Do you expect this to be used as a "public" constant? If not, consider prefix with an underscore.

https://google.github.io/styleguide/pyguide.html#3162-naming-conventions

Reply:
Thanks for raising this. Curious what you think about this...

This is in a private module (none of its objects are included in the reference docs), so my sense is that a leading underscore would be redundant. This is the pattern we've tended toward in the v2 codebase (e.g., private functions tend to not feature a leading underscore).

![image](https://github.com/kubeflow/pipelines/assets/55268212/ef2632aa-464e-4346-abbf-35cad3ae3b74)

I opted to not strip the leading underscore from `_DEFAULT_BASE_IMAGE` since it is currently unused outside of this module, so it is in a sense "private" to the module.

Not a strong feeling at all, so happy to make a change if desired.

Reply:
Makes sense to keep both as-is.